File: exercise/kotlin/day07/src/test/kotlin/CapturingLogger.kt
import ci.Logger


data class CapturingLogger(val lines: MutableList<String> = mutableListOf()) : Logger {
    override fun info(message: String) {
        lines.add("INFO: $message")
    }

    override fun error(message: String) {
        lines.add("ERROR: $message")
    }
}

File: exercise/c#/Day07/Day07.Tests/Doubles/CapturingLogger.cs
using Day07.CI.Dependencies;

namespace Day07.Tests.Doubles
{
    public class CapturingLogger : ILogger
    {
        private readonly List<string> _lines = [];
        public void Info(string message) => _lines.Add($"INFO: {message}");
        public void Error(string message) => _lines.Add($"ERROR: {message}");
        public IReadOnlyList<string> LoggedLines => _lines;
    }
}